src/Microsoft.AspNet.Mvc.Core/OutputFormatterOptions.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNet.Mvc.Core
+{
+    public class OutputFormatterOptions
+    {
+        private Dictionary<string, MediaTypeHeaderValue> map = new Dictionary<string, MediaTypeHeaderValue>();
+
+        public void AddFormatMapping(string format, MediaTypeHeaderValue contentType)
+        {
+            if (!string.IsNullOrEmpty(format) && contentType != null)
+            {
+                if(format.StartsWith("."))
+                {
+                    format = format.TrimStart('.');
+                }
+                                
+                map[format.ToLower()] = contentType;
+            }
+        }
+
+        public MediaTypeHeaderValue GetContentTypeForFormat(string format)
+        {
+            if (!string.IsNullOrEmpty(format))
+            {
+                if (format.StartsWith("."))
```

Comment:
You could simply call `TrimStart` here. Although you might want to be specify and truncate just the first '.' char depending on what your contract is. For instance is `.....foo` a supported format?